### PR TITLE
add printer unit test for DRA resources

### PR DESCRIPTION
**Pull request by googs1025**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
- add printer ut for DRA resources
#### Which issue(s) this PR fixes:

Fixes # None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Comment by googs1025** [comment]
@pohly  If you have time, please help take a look.

**Comment by BenTheElder** [comment]
/retitle add printer unit test for DRA resources

(hopefully a bit clearer for the git log)

**Comment by googs1025** [comment]
> /retitle add printer unit test for DRA resources
> 
> (hopefully a bit clearer for the git log)

Thanks for the reminder, it has been modified!

File: pkg/printers/internalversion/printers_test.go
Comment (toVersus) [inline pkg/printers/internalversion/printers_test.go:6410]: There are cases where the scheduler might not be able to determine which Node to schedule on, so it could be helpful to have a test case for when selectedNode is not set (i.e., column of `selectedNode` is an empty string).
https://github.com/kubernetes/enhancements/blob/334d04cd9fc677095f67d729f38d8d67df80d028/keps/sig-node/3063-dynamic-resource-allocation/README.md#resource-allocation-and-usage-flow

File: pkg/printers/internalversion/printers_test.go
Comment (googs1025) [reply pkg/printers/internalversion/printers_test.go:6372]: thanks. added

File: pkg/printers/internalversion/printers_test.go
Comment (googs1025) [reply pkg/printers/internalversion/printers_test.go:6434]: fixed

File: pkg/printers/internalversion/printers_test.go
Comment (googs1025) [reply pkg/printers/internalversion/printers_test.go:6221]: fixed

**Review by pohly — COMMENTED** [review]
Thanks for helping out here. It's important to improve unit test coverage because we want to promote to beta in 1.32.

Looks good to me, but I'll leave the detailed review to @toVersus .

File: pkg/printers/internalversion/printers_test.go
Comment (toVersus) [reply pkg/printers/internalversion/printers_test.go:6221]: @googs1025 

It seems like the revised commit hasn’t been pushed yet. Could you check on that?

Comment (googs1025) [reply pkg/printers/internalversion/printers_test.go:6221]: Thanks for your reply, I have pushed it up. I am used to using git commit --amend to merge commits with the same content. :)

Comment (toVersus) [reply pkg/printers/internalversion/printers_test.go:6221]: I often use `--amend` to overwrite commits too, but it looks like the amended commit hasn’t been pushed yet! This PR comment, like the others, hasn't been marked as `Outdated`...

Comment (googs1025) [reply pkg/printers/internalversion/printers_test.go:6221]: I changed all the changes to time.Now(), did I misunderstand? The review suggested changing it to time.Now().Add(1.9e9)?

Comment (googs1025) [reply pkg/printers/internalversion/printers_test.go:6221]: In addition, the other two add new test cases according to the review suggestions, without involving the old changes.

Comment (toVersus) [reply pkg/printers/internalversion/printers_test.go:6221]: > I changed all the changes to time.Now(), did I misunderstand?

Ah, sorry for the noise...🤦‍♂️
Since other existing test cases were using `time.Now().Add(1.9e9)`, I thought it would be better to align with them.

Comment (googs1025) [reply pkg/printers/internalversion/printers_test.go]: yes, that sounds reasonable.
fixed

**Review by toVersus — APPROVED** [review]

**Comment by toVersus** [comment]
/assign @ardaguclu

Can you take a look at this?

File: pkg/printers/internalversion/printers_test.go
Comment (ardaguclu) [inline pkg/printers/internalversion/printers_test.go:6220]: Could you please set here something other than `deviceclass` to not confuse with the resource name?

File: pkg/printers/internalversion/printers_test.go
Comment (ardaguclu) [inline pkg/printers/internalversion/printers_test.go:6221]: How do we guarantee that `time.Now().Add(1.9e9)` will always be translated to `0s` to prevent any flakes?.

File: pkg/printers/internalversion/printers_test.go
Comment (ardaguclu) [inline pkg/printers/internalversion/printers_test.go:6235]: Could you please add a test case for Empty creation timestamp that is translated to `<unknown>`?

File: pkg/printers/internalversion/printers_test.go
Comment (ardaguclu) [inline pkg/printers/internalversion/printers_test.go:6346]: Is it possible to manage human duration to put other than `0s`?

Comment (googs1025) [reply pkg/printers/internalversion/printers_test.go:6346]: I will change the result of all Age fields to 5m, which may be easier to understand.

File: pkg/printers/internalversion/printers_test.go
Comment (googs1025) [reply pkg/printers/internalversion/printers_test.go]: I have added in all cases

File: pkg/printers/internalversion/printers_test.go
Comment (googs1025) [reply pkg/printers/internalversion/printers_test.go]: I have added the "test-" prefix to all the cases.

**Comment by ardaguclu** [comment]
That looks good to me, thank you. I'll defer `lgtm` to @toVersus. After that I can gladly approve this PR.

**Review by toVersus — APPROVED** [review]
@ardaguclu
Thanks for having the time to review this! I'm not a collaborator, so I don't have permission to `/lgtm` the PR. Could you `/lgtm` it as well?

**Pull request merged**
